### PR TITLE
StatusLineManager: improve error with callstack

### PR DESCRIPTION
**Pull request by jukzi**
https://github.com/eclipse-jdt/eclipse.jdt.ui/issues/61

### CODE REVIEW

**Comment by jukzi** [comment]
@iloveeclipse splitted from https://github.com/eclipse-platform/eclipse.platform.ui/pull/1100#discussion_r1322609274

File: bundles/org.eclipse.jface/src/org/eclipse/jface/action/StatusLineManager.java [hunk 1]
```diff
@@ -149,14 +149,16 @@ public IProgressMonitor getProgressMonitor() {
...
 			public void beginTask(String name, int totalWork) {
-				if (taskStarted) {
-					throw new IllegalStateException("beginTask must only be called once per instance"); //$NON-NLS-1$
+				if (taskStarted != null) {
+					throw new IllegalStateException(
+							"beginTask should only be called once per instance. At least call done() before further invocations", //$NON-NLS-1$
+							taskStarted);
 				}
-				taskStarted = true;
+				taskStarted = new IllegalStateException("beginTask called first time"); //$NON-NLS-1$
```
Comment (laeubi) [inline bundles/org.eclipse.jface/src/org/eclipse/jface/action/StatusLineManager.java]:
```suggestion
				taskStarted = new IllegalStateException("beginTask("+name+", "+totalWork+") was called here previously"); 
```

**Review by laeubi — APPROVED** [review]
this should make it much easier to find who has called begin before...

File: bundles/org.eclipse.jface/src/org/eclipse/jface/action/StatusLineManager.java [hunk 2]
```diff
@@ -149,14 +149,16 @@ public IProgressMonitor getProgressMonitor() {
...
 			public void beginTask(String name, int totalWork) {
-				if (taskStarted) {
-					throw new IllegalStateException("beginTask must only be called once per instance"); //$NON-NLS-1$
+				if (taskStarted != null) {
+					throw new IllegalStateException(
+							"beginTask should only be called once per instance. At least call done() before further invocations", //$NON-NLS-1$
+							taskStarted);
 				}
-				taskStarted = true;
+				taskStarted = new IllegalStateException("beginTask("+name+", "+totalWork+") was called here previously");
```
Comment (iloveeclipse) [inline bundles/org.eclipse.jface/src/org/eclipse/jface/action/StatusLineManager.java]: This breaks compilation because of not externalized text

**Review by iloveeclipse — CHANGES_REQUESTED** [review]

File: bundles/org.eclipse.jface/src/org/eclipse/jface/action/StatusLineManager.java (hunk 2 above)
Comment (jukzi) [reply bundles/org.eclipse.jface/src/org/eclipse/jface/action/StatusLineManager.java]: done

**Review by iloveeclipse — APPROVED** [review]

**Pull request merged**